---
created: '2021-04-02T13:00:20.760130Z'
creator: sentry
source: tests/sentry/grouping/test_variants.py
---
app:
  hash: null
  component:
    app (exception of system takes precedence)
      exception (ignored because this variant does not have a contributing stacktrace, but the system variant does)
        stacktrace
          frame (non app frame)
            filename*
              "main.cpp"
            function*
              "main"
          frame (non app frame)
            filename*
              "main.cpp"
            function* (isolated function)
              "`anonymous namespace'::start"
          frame (non app frame)
            filename*
              "main.cpp"
            function* (isolated function)
              "`anonymous namespace'::crash"
          frame (non app frame)
            filename*
              "main.cpp"
            function* (isolated function)
              "`anonymous namespace'::something::nested::Foo"
        type (ignored because exception is synthetic)
          "EXC_BAD_ACCESS / KERN_INVALID_ADDRESS"
      threads (thread has no stacktrace)
--------------------------------------------------------------------------
system:
  hash: "85da85d9a2ef4672c4c4137f9413785c"
  component:
    system*
      exception*
        stacktrace*
          frame*
            filename*
              "main.cpp"
            function*
              "main"
          frame*
            filename*
              "main.cpp"
            function* (isolated function)
              "`anonymous namespace'::start"
          frame*
            filename*
              "main.cpp"
            function* (isolated function)
              "`anonymous namespace'::crash"
          frame*
            filename*
              "main.cpp"
            function* (isolated function)
              "`anonymous namespace'::something::nested::Foo"
        type (ignored because exception is synthetic)
          "EXC_BAD_ACCESS / KERN_INVALID_ADDRESS"
